lang/java/reef-webserver/src/main/java/org/apache/reef/webserver/HttpServerReefEventHandler.java

```diff
@@ -169,17 +170,17 @@ public void onHttpRequest(
       final List names = parsedHttpRequest.getQueryMap().get("filename");
```

Comment:
names seems to be of List<String>. Can you change this to avoid the dynamic typecast?